### PR TITLE
adding support for no prompt region in main.tf

### PR DESCRIPTION
**Pull request by bernadinm**
Many users have been wanting to set their region in the main.tf. This will be a good reason to set this not in the environment variable as we move into multi-region efforts as we do not want to set all remote regions in environment variables.
quickstart/docs: adding support for no prompt region

### CODE REVIEW

**Comment by fatz** [comment]
yeah this is a valid way of doing it and we need it for multi-region anyways.

one more point to this is explicitly setting the provider in the module call https://github.com/dcos-terraform/terraform-aws-dcos/blob/master/main.tf#L128-L130

you should do a test by setting your local defaults to AWS_DEFAULT_REGION to a different zone than in the provider call.

File: docs/published/main.tf [hunk 1]
```diff
@@ -11,6 +15,10 @@ module "dcos" {
   source  = "dcos-terraform/dcos/aws"
   version = "~> 0.1"
 
+  providers = {
+    aws = "aws"
+  }
+
```
Comment (fatz) [inline docs/published/main.tf]: ok perfect the example file has this already included. It should also be included in the code example above

**Comment by bernadinm** [comment]
```variable "dcos_install_mode" {
  description = "specifies which type of command to execute. Options: install or upgrade"
  default     = "install"
}

data "http" "whatismyip" {
  url = "http://whatismyip.akamai.com/"
}

provider "aws" {
  region = "us-east-1"
}

provider "aws" {
  region = "us-west-2"
  alias = "remote"
}

module "dcos1" {
  source  = "dcos-terraform/dcos/aws"
  version = "~> 0.1"

  providers = {
    aws = "aws"
  }

  dcos_instance_os    = "coreos_1855.5.0"
  cluster_name        = "region1"
  ssh_public_key_file = "~/.ssh/id_rsa.pub"
  admin_ips           = ["${data.http.whatismyip.body}/32"]

  num_masters        = "1"
  num_private_agents = "2"
  num_public_agents  = "1"

  dcos_version = "1.11.4"

  # dcos_variant              = "ee"
  # dcos_license_key_contents = "${file("./license.txt")}"
  dcos_variant = "open"

  dcos_install_mode = "${var.dcos_install_mode}"
}

module "spoke-1" {
  source  = "git@github.com:bernadinm/terraform-aws-remote-agents"
  #  version = "~> 0.1"

  providers = {
    aws   = "aws.remote"
  }

  dcos_instance_os    = "coreos_1855.5.0"
  cluster_name        = "region2"
  ssh_public_key_file = "~/.ssh/id_rsa.pub"
  admin_ips           = ["${data.http.whatismyip.body}/32"]

  num_private_agents = "2"
  num_public_agents  = "1"

  subnet_range = "172.14.0.0/16"

  dcos_install_mode = "${var.dcos_install_mode}"
}

output "masters-ips-site1" {
  value = "${module.dcos1.masters-ips}"
}

output "cluster-address-site1" {
  value = "${module.dcos1.masters-loadbalancer}"
}

output "public-agents-loadbalancer-site1" {
  value = "${module.dcos1.public-agents-loadbalancer}"
}

output "private-agents-ips-site2" {
  value = "${module.spoke-1.private_agents-ips}"
}

output "public-agents-ips-site2" {
  value = "${module.spoke-1.public_agents-ips}"
}

output "public-agents-loadbalancer-site2" {
  value = "${module.spoke-1.public-agents-loadbalancer}"
}
```

File: docs/published/main.tf (hunk 1 above)
Comment (fatz) [reply docs/published/main.tf]: @bernadinm thanks for the example. This request for change was to update .md files to contain this example.

Comment (bernadinm) [reply docs/published/main.tf]: ah! thank you!

Comment (fatz) [reply docs/published/main.tf]: @bernadinm still missing in published/README.md

**Pull request merged**